Assets/Scripts/Player/WeaponInventory.cs

```diff
@@ -12,14 +12,10 @@ public class WeaponInventory : MonoBehaviour
     public int MaxWeapons = 3;
     public GameObject Pickupable;
 
-    [Reorderable]
-    public ListOfWeapons Weapons;
-
-    [System.Serializable, HideInInspector]
-    public class ListOfWeapons : ReorderableArray<WeaponData>
-    {
-    }
+    public List<WeaponData> NewWeapons;
+    private static List<WeaponData> Weapons = new List<WeaponData>();
 
+    private bool hasAddedWeapons = false;

```

Comment:
Can you explain what this is used for?

Reply:
Its to avoid `Start()` repeatable adding weapons each time the player moves through   scene

Reply:
Would this be fixed if we used `Awake()` instead of `Start()`? Or is this the best option?

Reply:
No it wouldn't change the scenes

Reply:
👍